control/prometheus.py

```diff
@@ -0,0 +1,284 @@
+#
+#  Copyright (c) 2021 International Business Machines
+#  All rights reserved.
+#
+#  SPDX-License-Identifier: LGPL-3.0-or-later
+#
+#  Authors: pcuzner@ibm.com
+#
+
+import os
+import time
+import logging
+import threading
+import spdk.rpc as rpc
+
+from prometheus_client.core import REGISTRY, GaugeMetricFamily, CounterMetricFamily, InfoMetricFamily
+from prometheus_client import start_http_server
+from typing import NamedTuple
+
+
+class RBD(NamedTuple):
+    pool: str
+    namespace: str
+    image: str
+
+
+def timer(method):
```

Comment:
I'd normally think of summaries metrics for quantiles like histograms at least they do for the other clients - so I don't want to adopt them here since things would change as and when the python client catches up with java/golang etc.

Reply:
In the example they expressly use Summary for measuring time elapsed. I agree that the metric names chosen are perhaps confusing (I find this a common trait in the Prometheus project).